### PR TITLE
Kmer Index Migration

### PR DESCRIPTION
**Pull request by pjbradbury**
## Description

_Provide a summary of your changes including motivation and context.
If these changes fix a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated relevant documentation

## Changelog entry

_Please add a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

**Review by zrm22 — APPROVED** [review]

**Review by lynnjo — APPROVED** [review]

File: src/test/kotlin/net/maizegenetics/phgv2/pathing/BuildKmerIndexTest.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                File(hvcfFile).copyTo(dst)
+            }
+        }
+
+        //create a HaplotypeGraph from the hvcf files
+        val buildIndexResult = BuildKmerIndex().test("--agc-path $tempAGCDir --hvcf-dir $tempHvcfDir")
+
+        //Was the index created?
+        assertEquals(0, buildIndexResult.statusCode)
+        assert(File("${tempHvcfDir}/kmerIndex.txt").exists())
```
[the remark this answers is not in the document]
Comment (pjbradbury) [reply src/test/kotlin/net/maizegenetics/phgv2/pathing/BuildKmerIndexTest.kt:84]: I could not come up with a good way to test the kmer index contents directly that would be better than (or even as good as) round tripping the file once the reader is finished.

File: build.gradle.kts [hunk 2]
```diff
@@ -41,6 +41,8 @@ dependencies {
 
     implementation("com.github.samtools:htsjdk:4.0.1")
 
+    implementation("it.unimi.dsi:fastutil:8.5.12")
```
Comment (tcasstevens) [inline build.gradle.kts:44]: You may want to sync with main as I recently merged a change to build.gradle.kts

File: src/main/kotlin/net/maizegenetics/phgv2/api/HaplotypeGraph.kt [hunk 3]
```diff
@@ -31,7 +31,7 @@ class HaplotypeGraph(hvcfFiles: List<String>) {
     private lateinit var refRangeMap: SortedMap<ReferenceRange, Int>
 
     // Map<ID (checksum), AltHeaderMetaData>
-    private val altHeaderMap: MutableMap<String, AltHeaderMetaData> = mutableMapOf()
```
Comment (tcasstevens) [inline src/main/kotlin/net/maizegenetics/phgv2/api/HaplotypeGraph.kt:34]: I have a method for this now. I think this should be put back to private

**Review by tcasstevens — APPROVED** [review]

File: src/main/kotlin/net/maizegenetics/phgv2/api/HaplotypeGraph.kt (hunk 3 above)
Comment (tcasstevens) [reply src/main/kotlin/net/maizegenetics/phgv2/api/HaplotypeGraph.kt:34]: Might want to wait until I merge in my HaplotypeGraph branch

**Pull request merged**